nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/impl/TestPropertyCommentManager.java

```diff
@@ -0,0 +1,399 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ *     Nuno Cunha <ncunha@nuxeo.com>
+ */
+
+package org.nuxeo.ecm.platform.comment.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.nuxeo.ecm.platform.comment.api.Comments.commentToDocumentModel;
+import static org.nuxeo.ecm.platform.comment.api.Comments.newComment;
+import static org.nuxeo.ecm.platform.comment.workflow.utils.CommentsConstants.COMMENT_DOC_TYPE;
+
+import java.time.Instant;
+import java.util.Calendar;
+import java.util.List;
+
+import org.junit.Test;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.PathRef;
+import org.nuxeo.ecm.platform.comment.AbstractTestCommentManager;
+import org.nuxeo.ecm.platform.comment.api.Comment;
+import org.nuxeo.ecm.platform.comment.api.CommentImpl;
+import org.nuxeo.runtime.test.runner.Deploy;
+
+/**
+ * @since 10.3
+ */
+@Deploy("org.nuxeo.ecm.platform.query.api")
+public class TestPropertyCommentManager extends AbstractTestCommentManager {
+
+    @Test(expected = NuxeoException.class)
```

Comment:
Testing exception case is really good, but `NuxeoException` can be thrown in many places and so this assertion is not strong enough. Furthermore don't you add/rework such tests in `AbstractTestCommentManager` ?

Reply:
Yes, you can check the details of the exception (error code and message).

The other class (AbstractTestCommentManager) also has unit tests, but I've changed the minimum possible there in order to reduce the scope of this change and avoid compatibility issues with other CommentManager implementations.
Anyway, I would be happy if we can rework that in order to have separated tests per scenario and, probably, per implementation. 👍